src/machine/machine_atmega.go

```diff
@@ -155,3 +155,94 @@ func (uart UART) WriteByte(c byte) error {
 	avr.UDR0.Set(c) // send char
 	return nil
 }
+
+// SPIConfig is used to store config info for SPI.
+type SPIConfig struct {
+	Frequency uint32
+	SCK       Pin
+	SDO       Pin
+	SDI       Pin
+	LSBFirst  bool
+	Mode      uint8
+}
+
+// SPI is for the Serial Peripheral Interface
+// Data is taken from http://ww1.microchip.com/downloads/en/DeviceDoc/ATmega48A-PA-88A-PA-168A-PA-328-P-DS-DS40002061A.pdf page 169 and following
+type SPI struct{}
+
+// Configure is intended to setup the SPI interface.
+func (spi SPI) Configure(config SPIConfig) error {
+
+	// Make the defaults meaningful
+	if config.Frequency == 0 {
+		config.Frequency = 4000000
+	}
+	if config.SCK == 0 {
+		config.SCK = SPI0_SCK_PIN
+	}
+	if config.SDO == 0 {
+		config.SDO = SPI0_SDO_PIN
+	}
+	if config.SDI == 0 {
+		config.SDI = SPI0_SDI_PIN
+	}
```

Comment:
This doesn't make sense to me. AVR chips have assigned fixed pins to certain SPI peripherals, so there is no point in configuring different pins.

Reply:
I based this on the prior PR as well as trying to be consistent with the other SPI implementations for consistency.  I'm fine deleting code that isn't used.